File: holidays-service/src/main/java/dev/enzosantos/timesheet/infrastructure/persistence/mapper/HolidayMapper.java
package dev.enzosantos.timesheet.infrastructure.persistence.mapper;

import dev.enzosantos.timesheet.domain.entities.Holiday;
import dev.enzosantos.timesheet.infrastructure.persistence.jpa.HolidayEntity;

public class HolidayMapper {
    public static HolidayEntity toEntity(Holiday domain) {
        HolidayEntity entity = new HolidayEntity();
        entity.setDate(domain.date());
        entity.setScope(domain.scope());
        entity.setName(domain.name());
        entity.setType(domain.type());
        return entity;
    }

    public static Holiday toDomain(HolidayEntity entity) {
        return new Holiday(entity.getDate(), entity.getName(), entity.getType(), entity.getScope());
    }
}

File: users-service/src/infrastructure/db/mappers/holiday.mapper.ts
import { Holiday } from "../../../domain/entities/holiday"
import { HolidayEntity } from "../entities/holiday.entity"
import { HolidayShiftEntity } from "../entities/holiday-shift.entity"

export namespace HolidayMapper {
    export function toDomain(entity: HolidayEntity): Holiday {
        return {
            id: entity.id,
            name: entity.name,
            date: entity.date,
            shift: entity.shift,
        }
    }

    export function toEntity(domain: Holiday): HolidayEntity {
        const entity = new HolidayEntity()
        entity.id = domain.id
        entity.name = domain.name
        entity.date = domain.date
        entity.shift =
            domain.shift == null
                ? null
                : {
                      am: HolidayShiftEntity.am,
                      pm: HolidayShiftEntity.pm,
                  }[domain.shift]
        return entity
    }
}
